daemon/daemon_unix.go

```diff
@@ -472,28 +472,28 @@ func verifyContainerResources(resources *containertypes.Resources, sysInfo *sysi
 func (daemon *Daemon) getCgroupDriver() string {
 	cgroupDriver := cgroupFsDriver
 
-	// No other cgroup drivers are supported at the moment. Warn the
-	// user if they tried to set one other than cgroupfs
-	for _, option := range daemon.configStore.ExecOptions {
+	if daemon.usingSystemd() {
+		cgroupDriver = cgroupSystemdDriver
```

Comment:
Sorry, I meant checking that the driver name is valid. I agree that the dbus call is a bit expensive to have twice.

atm that would be either  "cgroupfs" or "systemd". 

If a user mistyped and pass `systmd` he would be none the wiser.

Actually, maybe we should just error out if the passed name is invalid. WDYT?